crates/metal-mir/src/structure.rs

```diff
@@ -0,0 +1,38 @@
+use crate::types::{visibility::Visibility, Type};
+
+/// Represents the properties of a struct.
+/// i.e. `a: B`
+#[derive(Debug, Clone)]
+pub struct StructProperty {
+    /// name of the struct property. i.e.: `a`
+    pub name: &'static str,
+    /// index of the struct property (order,) i.e. `0`
+    pub idx: u32,
+    /// the type of this struct property. i.e. `String`
+    pub ty: Type,
+}
+
+/// Represents a struct in Metal.
+/// i.e.:
+/// ```metal
+/// pub struct A {
+///     b: String;
+///     pub c: i32;
```

Comment:
I believe these should be commas?

Reply:
definitely! already fixed it locally. This was when we were still doing the functions-in-structs and enums thing, so `;` was mostly there since it felt better for properties. Since it's not like that, I'll just change it to a comma.